numpy/core/src/umath/matmul.c.src

```diff
@@ -506,6 +558,11 @@ NPY_NO_EXPORT void
 
 #endif
     }
+#if @USEBLAS@ && defined(HAVE_CBLAS)
+    free(tmp_ip1);
+    free(tmp_ip2);
+    free(tmp_op);
```

Comment:
Checking for NULL would save a syscall.

Reply:
Checked `free` on https://godbolt.org and indeed, it always creates a syscall, even with the latest GCC and Clang and `-O3`. Quite disappointing. I've thus added checks for NULL.